src/esp/physics/bullet/BulletPhysicsManager.cpp

```diff
@@ -272,13 +310,93 @@ int BulletPhysicsManager::getNumActiveContactPoints() {
         static_cast<const btCollisionObject*>(manifold->getBody1());
 
     // logic copied from btSimulationIslandManager::buildIslands. We want to
-    // count manifold points only if related to non-sleeping bodies.
+    // count manifolds only if related to non-sleeping bodies.
     if (((colObj0) && colObj0->getActivationState() != ISLAND_SLEEPING) ||
         ((colObj1) && colObj1->getActivationState() != ISLAND_SLEEPING)) {
-      pointCount += manifold->getNumContacts();
+      count += manifold->getNumContacts();
+    }
+  }
+  return count;
+}
+
+std::vector<ContactPointData> BulletPhysicsManager::getContactPoints() const {
+  if (m_recentNumSubStepsTaken != 1) {
+    if (m_recentNumSubStepsTaken == -1) {
+      LOG(WARNING) << "getContactPoints: no previous call to stepPhysics";
+    } else {
+      // todo: proper logging-throttling API
+      static int count = 0;
```

Comment:
This static local is super hacky. And this is a ton of code here just to avoid spamming this warning every frame. I didn't intend this code to merge to master as-is when I wrote it. As I've said before, we should be aiming to clean up these hacks as we're merging AO to master.

Reply:
Maybe we don't log anything here and just document the function. This only scrapes contacts from the most recent substep or call to discrete collision detection, right? That is pretty straightforward. I guess we could also add a query for the `recentNumSubStepsTaken_` so users can pivot or check this manually if they want to.